src/List.js

```diff
@@ -66,13 +86,33 @@ export function List() {
 
   const handleChange = async (id, event) => {
     let date = new Date();
+    const item = items.find((element) => element.id === id);
+    const daysSinceLastTransaction = item.lastPurchasedDate
```

Comment:
`.find` can return undefined, meaning that `item` could be undefined. If that was the case then we would get a null pointer error here. We can avoid this by doing `item?.lastPurchasedDate`.